### PR TITLE
Update base dependencies

### LINKED ISSUE #347: Only define base as peer dependency
# Issue Description
Terra components have been declaring terra-base as a dependency and a peer dependency. We should move this to be a peer dependency to ensure only one version is installed and referenced.

See discussion here: https://github.com/cerner/terra-core/pull/1583/files

Sync https://github.com/cerner/terra-core/issues/1587

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [ ] Bug
- [x] Other

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Closes #347

### CODE REVIEW

File: packages/terra-clinical-application/package.json [hunk 1]
```diff
@@ -33,7 +33,6 @@
   "dependencies": {
     "classnames": "^2.2.5",
     "prop-types": "^15.5.8",
-    "terra-base": "^3.0.0",
```
Comment (StephenEsser) [inline packages/terra-clinical-application/package.json]: Do we want to bump the peerDependency to the latest terra-base version? Should lerna be doing that automatically?

Comment (emilyrohrbough) [reply packages/terra-clinical-application/package.json]: It appears lerna hasn't been bumping peer dependencies 😞

Comment (StephenEsser) [reply packages/terra-clinical-application/package.json]: I wonder if it will only bump the peer dependency if they are in the same monorepo

Comment (bjankord) [reply packages/terra-clinical-application/package.json]: I believe lerna does not handle bumping peerDeps. We have to do that manually. I think leaving this at `^3.0.0` is fine or setting it to `^3.12.0` is also fine. Both pull in the latest minor version of v3, its just a diff in what the lower bound version be. As long as the v3 minor and patch releases are following semver, using `^3.0.0` for the peerDep shouldn't be an issue.

**Pull request merged**